content/blog/vim-getting-started/index.md

```diff
@@ -0,0 +1,138 @@
+---
+title: "Vim : Getting started"
+date: "2020-10-08"
+author: "Shubhankar Khare"
+coverImage: "Vim.png"
+tags: ["Vim"]
+description: "A guide to getting you started with VIM: What is VIM, Why is VIM,  How to configure VIM."
+---
+
+## Vim - Getting started
+Before VsCode, Sublime Text, Atom or any other modern text editor, humankind was blessed by Bill Joy who created **Vi** in 1976. Vi was designed to work on terminals and with the very uncommon characteristic of working in a **modal fashion**.
+
+**Vim** (**V**i **IM**proved) is the latest and most celebrated reincarnation of Vi which supports both text and graphical interfaces and is supported on every platform known.
+
+#### Why the big Fuss?
+Remember I said Modal fashion, well it's these modes **( Normal, Insert and Visual)**  in Vim which makes it such a powerful text editor. These **modes give you a completely different level of control and fluency when editing code.**
+
+Also, It is **extremely customizable** and you can adapt it to your way of coding and your way of doing things.
+To give you a taste of vim , below are a few commands frequently used to code at lightning speed:
+
+ - hjkl : move the cursor around to left, down, up, and right respectively.
+ - 7j : move 7 lines down.
+ - w : move a word forward
+ - ctrl + f : move down a page
+ - ctrl + b : move up a page
+ - gg : move to the top of the document
+ - G : move to the bottomost of the document
+ - dw : delete a word
+ - d6w : delete 6 words
+ - dt> : delete till  `>`
+ - di] : delete everything inside `[ ]`
+ - dd : delete whole line
+ - 4dd : delete 4 lines
+ - yy : yank a line ( yank is copy )
+ - cc : change a line ( change is delete and go in insert mode )
+ - cap : change a paragraph
+ - `.` : repeat last command
+ - f' : find first occurance of `'`
+ - f'ci'hello : _find the next  `'` then change everything inside `'` for `hello`
+
+The list goes on 
+
+ So basically we have standard keybindings like d for delete, c for change, f for find etc. and we can pair them with numbers to repeat that action n number of times. Moreover doubling a command like so `yy` makes the command operate on the entire line.
+
+
+So to sum up,
+>Vim is for **programmers who want to raise their game**. In the hands of an expert, It shreds text at the speed of thought.
+
+ #### Cool, how do I get started?
+ Well, there is a learning curve, but it's worth it. Although this steep curve is exaggerated in the online world and the **quitting vim joke** is getting very old ( It's:q btw ).
+
+ Install Vim `sudo apt-get vim or pacman -S vim` for Linux user and type `vim <filename>` in your terminal. Press `i` to go to insert mode. Type some gibberish. Press `:wq` to save and quit . `:q!` to save without quitting.
```

Comment:
@iamshubhankarkhare don't you think `!q` is for quitting without saving? Just wanted to clear the doubt here.